Modules/QuestieValidateGameCache.lua

```diff
@@ -0,0 +1,145 @@
+    -- After MAX_INIT_QUEST_LOG_TRIES tries we stop trying
+    if numEntries == 0 then
+        initQuestLogTries = initQuestLogTries + 1
+        if initQuestLogTries < MAX_INIT_QUEST_LOG_TRIES then
+            print("--> QuestieValidateGameCache.OnQuestLogUpdate()", GetTime(), "Game's quest log is empty. Tries:", initQuestLogTries.."/"..MAX_INIT_QUEST_LOG_TRIES)
+            _WaitForGameCache_TryAgainAtEvent()
+            return
+        else
+            print("--> QuestieValidateGameCache.OnQuestLogUpdate()", GetTime(), "Decide player has no quests for real.")
+        end
+    end
+    ]]--
+
+    local isGameCacheGood = true
+    local goodQuestsCount = 0 -- for debug stats
+
+    for i = 1, MAX_QUEST_LOG_INDEX do
+        local title, _, _, isHeader, _, _, _, questId = GetQuestLogTitle(i)
+        if (not title) then
+            break -- We exceeded the valid quest log entries
+        end
+        if (not isHeader) then
+            local hasInvalidObjective -- for debug stats
+            local objectiveList = GetQuestObjectives(questId)
+            for _, objective in pairs(objectiveList) do -- objectiveList may be {} and that is validly cached quest in game log
+                if (not objective.text) or stringSub(objective.text, 1, 1) == " " then
+                    -- Game hasn't cached the quest fully yet
+                    isGameCacheGood = false
+                    hasInvalidObjective = true
+
+                    -- No early "return false" here to force iterate whole quest log and speed up caching
+                end
+            end
+            if not hasInvalidObjective then
+                goodQuestsCount = goodQuestsCount + 1
+            end
+        end
+    end
+
+    if not isGameCacheGood then
+        print("--> QuestieValidateGameCache.OnQuestLogUpdate()", GetTime(), "Game's quest log is not yet okey. Good quest: "..goodQuestsCount.."/"..numQuests)
+        return
+    end
+
+    if goodQuestsCount ~= numQuests then
+        -- This shouldn't be possible
+        Questie:Error("Report this error! Game QuestLog cache is broken. Good quest: "..goodQuestsCount.."/"..numQuests) -- TODO fix message and add translations? translations might not be available?
+        -- TODO should we stop whole addon loading progress?
+    end
+
+    print("--> QuestieValidateGameCache.OnQuestLogUpdate()", GetTime(), "Game's quest log is |cff00bc32".."OK".."|r. Good quest: "..goodQuestsCount.."/"..numQuests)
+
+    DestroyEventFrame()
+
+    cacheGood = true
+
+    -- Call all callbacks
+    while (#callbacks > 0) do
+        local callback = table.remove(callbacks, 1)
+        local func, args = tunpack(callback)
+        print("--> QuestieValidateGameCache.OnQuestLogUpdate()", "Calling a callback.")
+        func(tunpack(args))
+    end
+end
+
+local function OnPlayerEnteringWorld(_, _, isInitialLogin, isReloadingUi)
+    assert(isInitialLogin or isReloadingUi)
+
+    numberOfQuestLogUpdatesToSkip = isReloadingUi and 0 or 1
```

Comment:
```suggestion
    numberOfQuestLogUpdatesToSkip = isReloadingUi and 0 or 1 -- We want to skip the first QUEST_LOG_UPDATE at login (fires twice at login and only once when reloading)
```

Reply:
It may fire many more times. But on login the first QUEST_LOG_UPDATE after PLAYER_ENTERING_WORLD has still invalid cache.

Reply:
Done in 78b16dab24437ff7c33ec1111cd51a01e38c7ea2 :
Logic changed to show better what we really want to do (functionality is still the same on current game behavior).
Comment added.